pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,390 @@
+        else:
+            raise ValueError(
+                f"Invalid value '{value}' scattering parameter requested for {self._manu} {self._model}. Valid values are: {SCATTERING_PARAMETERS}"
+            )
+
+    IFBW = Instrument.control(
+        "IFBW?",
+        "IFBW%d",
+        f"""Control the IF Bandwidth of the instrument for a scan sweep. 
+        (int from the set [{[f'"{value}"' for value in ALLOWED_BANDWIDTH]}]).""",
+        cast=lambda x: int(float(x)),
+        validator=strict_discrete_set,
+        values=ALLOWED_BANDWIDTH,
+    )
+
+    def reset(self):
+        """Reset the instrument. May cause RF Output power to be enabled!"""
+        self.write("*RST")
+        sleep(0.25)
+
+    def scan(self, timeout=10):
+        """Initiates a scan with the number of averages specified and
+        blocks until the operation is complete.
+
+        Args:
+        timeout - a value in seconds to add to the sweep time to timeout the function
+        """
+
+        # get time to perform sweep
+        sweep_time = self.sweep_time + timeout
+
+        # get number of averages if enabled
+        if self.averaging_enabled:
+            sweep_time = sweep_time * self.averages
+
+        self.scan_single()
+
+        # create a time limit
+        start = now()
+
+        # All queries will block until the scan is done, so use NOOP? to check.
+        # These queries will time out after several seconds though,
+        # so query repeatedly until the scan finishes.
+        while True:
+            try:
+                # status = self.ask("NOOP?")
+                self.ask("NOOP?")
+                break
+            except VisaIOError as e:
+                if e.abbreviation != "VI_ERROR_TMO":
+                    raise e
+            # calculate time sweep should be complete by
+            if now() > (start + sweep_time):
+                raise TimeoutError(
+                    f"VNA Scan took longer than {sweep_time} seconds to complete and timed out."
+                )
+
+    def scan_single(self):
+        """Initiates a single scan or N scans averaged based on averaging
+        This function is not blocking
+        """
+        if self.averaging_enabled:
+            self.write(f"NUMG{self.averages}")
+        else:
+            self.write("SING")
+
+    @property
+    def frequencies(self):
+        """Returns a list of frequencies from the last scan.
+        Does not communicate with instrument. (np.ndarray of
+        frequencys sized by number of points in sweep)"""
+        return np.linspace(self.start_frequency, self.stop_frequency, num=self.scan_points)
+
+    @property
+    def data_complex(self, timeout=10):
+        """Gets the complex s-parameter measurements from the last scan.
+        This function is blocking until it is completed. (returns nparray
+        sized by number of points in sweep)
+        
+        Args:
+        timeout - a value in seconds to timeout the function
```

Comment:
We use sphinx style to indicate parameters:
```suggestion
        :param timeout: timeout of the function in seconds
```
between ":param" and the parameter name, you can add the type, if you wish: ":param float timeout:".

Reply:
Fixed doc string to use the sphinx style.